addOns/client/src/main/java/org/zaproxy/addon/client/internal/ClientSideComponent.java

```diff
@@ -124,4 +122,57 @@ public boolean equals(Object obj) {
                 && Objects.equals(tagName, other.tagName)
                 && Objects.equals(text, other.text);
     }
+
+    @Override
+    public int compareTo(ClientSideComponent other) {
+        int result = 0;
+        result = stringCompare(this.getTypeForDisplay(), other.getTypeForDisplay());
+        if (result != 0) {
+            return result;
+        }
+        // Types are the same, check next relevant property(ies)
+        result = stringCompare(this.href, other.href);
+        if (result != 0) {
+            return result;
+        }
+        result = stringCompare(this.text, other.text);
+        if (result != 0) {
+            return result;
+        }
+        result = stringCompare(this.id, other.id);
+        if (result != 0) {
+            return result;
+        }
+        result = stringCompare(this.tagName, other.tagName);
+        if (result != 0) {
+            return result;
+        }
+        result = stringCompare(this.tagType, other.tagType);
+        if (result != 0) {
+            return result;
+        }
+        if (this.formId > other.formId) {
+            return 1;
+        } else if (this.formId < other.formId) {
+            return -1;
+        }
+        return 0;
+    }
+
+    private static int stringCompare(String one, String two) {
+        if (one == null || two == null) {
+            return nullCompare(one, two);
+        }
+        return one.compareTo(two);
+    }
+
+    private static int nullCompare(Object here, Object other) {
```

Comment:
Be good to use the same variable names for consistency.

Reply:
Isn't "this" a reserved word?

Reply:
I was referring to one two here other.

Reply:
Okay, got ya